File: Curso-ciencias-de-la-computacion/0_Problems/1_shiftZerosToEnd/C/shiftZerosToEnd.c
#include <stdio.h>

#define SIZE_ARRAY 5

void shift_zeros_to_end(int* array){
    int count = 0;
    int temp = 0;

    for (int i = 0; i < SIZE_ARRAY; i++){
        if (array[i] != 0){
            temp = array[i];
            array[i] = array[count];
            array[count] = temp;
            count++;
        }       
    }
}

void printArray(int* array){
    printf("The array is: ");
    for (int i = 0; i < SIZE_ARRAY; i++){
        printf("%d ", array[i]);
    }
}


int main(){
    int array[SIZE_ARRAY] = {0, 1, 0, 3, 5};

    shift_zeros_to_end(array);
    printArray(array);
    
    return 0;
}

File: Curso-ciencias-de-la-computacion/0_Problems/1_shiftZerosToEnd/Python/shift_zeros_to_end.py
def shiftZerosToEnd(array):
    count = 0
    for i in range(0, len(array)):
        if array[i] != 0:
            temp = array[i]
            array[i] = array[count]
            array[count] = temp
            count += 1

def main():
    array = [1, 0, 3, 0, 5, 0, 0, 77, 6, 0, 0, 0, 0, 1]
    shiftZerosToEnd(array)
    print(array)

if __name__ == '__main__':
    main()